File: hkoweather_bot/telegram.py
from __future__ import annotations

import requests


class TelegramClient:
    def __init__(self, token: str, chat_id: str) -> None:
        self._chat_id = chat_id
        self._base_url = f"https://api.telegram.org/bot{token}"

    def send_message(self, text: str) -> None:
        response = requests.post(
            f"{self._base_url}/sendMessage",
            json={
                "chat_id": self._chat_id,
                "text": text,
                "disable_web_page_preview": True,
            },
            timeout=20,
        )
        response.raise_for_status()

File: src/core/telegram.ts
import type { TelegramConfig } from "./types";

export async function sendTelegramMessage(config: TelegramConfig, text: string): Promise<void> {
  const response = await fetch(`https://api.telegram.org/bot${config.botToken}/sendMessage`, {
    method: "POST",
    headers: { "content-type": "application/json" },
    body: JSON.stringify({
      chat_id: config.chatId,
      text,
      disable_web_page_preview: true,
    }),
  });

  if (!response.ok) {
    throw new Error(`Telegram sendMessage failed with ${response.status}: ${await response.text()}`);
  }
}
